src/main/java/com/nha/abdm/wrapper/hip/hrp/database/mongo/services/PatientService.java

```diff
@@ -88,4 +93,64 @@ public boolean checkCareContexts(InitResponse data) {
     }
     return true;
   }
+
+  public AckRequest addPatient(LinkRecordsResponse data) {
+    String abhaAddress = data.getAbhaAddress();
+    try {
+      Patient existingRecord = this.patientRepo.findByAbhaAddress(abhaAddress);
+      if (existingRecord == null) {
+        throw new RuntimeException("Patient not found");
```

Comment:
why are throwing Runtime error? should we not add data to database if it is not present?

Reply:
Not throwing any runtime exceptions just loging them.